ipatests/test_integration/test_installation_client.py

```diff
@@ -71,10 +71,72 @@ def test_client_install_with_ssh_trust_dns(self):
 
         related: https://pagure.io/freeipa/issue/8082
         """
-        tasks.install_client(self.master, self.clients[0],
-                             extra_args=['--ssh-trust-dns'])
-        result = self.clients[0].run_command(['cat', '/etc/ssh/ssh_config'])
-        assert 'HostKeyAlgorithms' not in result.stdout_text
+        try:
+            tasks.install_client(self.master, self.clients[0],
+                                 extra_args=['--ssh-trust-dns'])
+            result = self.clients[0].run_command(['cat', '/etc/ssh/ssh_config'])
+            assert 'HostKeyAlgorithms' not in result.stdout_text
+        finally:
+            tasks.uninstall_client(self.clients[0])
+
+
+class TestClientInstallParams(IntegrationTest):
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_dns=True)
+
+    def test_client_install_sssd_params(self):
+        """
+        --permit
+        Configure  SSSD  to  permit all access.
+        Otherwise, the machine will be controlled
+        by the Host-based Access Controls (HBAC) on the IPA server.
+        """
+        client = self.clients[0]
+        tasks.install_client(self.master, client,
+                             extra_args=['--permit',
+                                         '--enable-dns-updates',
+                                         '--no-krb5-offline-passwords'])
+        try:
+            sssd_conf = client.transport.get_file_contents(paths.SSSD_CONF,
+                                                           encoding='utf-8')
+            exp_val_permit = "access_provider = permit"
+            assert exp_val_permit in sssd_conf
+
+            """
+            --enable-dns-updates
+            This option tells SSSD to automatically update DNS with
+            the  IP  address  of  this client.
+            """
+            exp_val_dyndns_update = "dyndns_update = True"
+            assert exp_val_dyndns_update in sssd_conf
+
+            """
+            --no-krb5-offline-passwords
+            Configure SSSD not to store user
+            password when the server is offline.
+            """
+            unexp_val_krb5_pass = "krb5_store_password_if_offline = True"
+            assert unexp_val_krb5_pass not in sssd_conf
+        finally:
+            tasks.uninstall_client(client)
+
+    def test_client_install_authconfig_params(self):
+        """
+        --mkhomedir
+        Configure PAM to create a users home directory
+        if it does not exist
+        """
+        client = self.clients[0]
+        tasks.install_client(self.master, client,
+                             extra_args=['--mkhomedir'])
+
+        authconf = client.get_file_contents('/etc/pam.d/system-auth',
+                                            encoding='utf-8')
+        exp_val_mkhomedir = "mkhomedir.so"
+        assert exp_val_mkhomedir in authconf
```

Comment:
Just a question: would it be more robust to call `authselect current` and parse that output so the test doesn't rely on a setting in a specific file?